edb/server/protocol/auth_ext/oauth.py

```diff
@@ -81,7 +82,43 @@ async def handle_callback(self, code: str) -> None:
     async def _handle_identity(
         self, user_info: data.UserInfo
     ) -> None:
-        ...
+        """Update or create an identity"""
+
+        await execute.parse_execute_json(
+            db=self.db,
+            query="""\
+with
+  iss := <str>$issuer_url,
+  sub := <str>$provider_id,
+  email := <optional str>$email,
+
+  # update an existing identity, if it exists
```

Comment:
why not use [our upsert](https://www.edgedb.com/docs/edgeql/insert#upserts) 😏

Reply:
Oh true, I originally had a much more complicated structure that had to look-up existing emails and find identities, etc. Lemme see if it can just be simplified back down to our `unless conflict` call.

Reply:
@fantix ⚡ [`c10bc1e` (#5941)](https://github.com/edgedb/edgedb/pull/5941/commits/c10bc1e05ca8a6e857f7ba7479f03d30c3b55125)

Thanks for noticing this!